src/GitHubExtension/Widgets/GitHubRepositoryWidget.cs

```diff
@@ -69,16 +70,35 @@ public override void OnActionInvoked(WidgetActionInvokedArgs actionInvokedArgs)
 
     protected override void ResetWidgetInfoFromState()
     {
+        JsonNode? dataObject = null;
         try
         {
-            var dataObject = JsonNode.Parse(ConfigurationData);
-
-            if (dataObject == null)
+            dataObject = JsonNode.Parse(ConfigurationData);
+        }
+        catch (JsonException)
+        {
+            // Old data versioning was not a Json string. If we attempt to parse
+            // and we get a failure, check if it is the old version.
+            if (Validation.IsValidGitHubURL(ConfigurationData))
             {
-                return;
+                Log.Logger()?.ReportInfo(Name, ShortId, $"Found string data format, migrating to JSON format. Data: {ConfigurationData}");
+                var migratedState = new JsonObject
+                {
+                    { "url", ConfigurationData },
+                };
+                ConfigurationData = migratedState.ToJsonString();

```

Comment:
Just wanted to double-check whether we need a try block here from line 80-95.

Reply:
Same question.  Otherwise code looks good

Reply:
Is this in case the Json object manipulation fails unexpectedly?

Reply:
ya.  I just don't know if it's needed.  Can ToJsonString() even fail like that?

Reply:
On second thought, yeah this is probably fine as is.

Reply:
I'm not sure, but since we've already had crashes in this area of the code, I added them out of sheer paranoia. We can take it out with a more robust migration solution.